docs/profiler.md

```diff
@@ -1,136 +1,196 @@
+Having run the tool, one may wonder, what does the output mean? Read on to understand the various input flags one can use to customize each profiler run and how to interpret the subsequent results.
 
-By default the profiler uses 1GiB sample data. This can be changed with the `--data-size` flag. Different sizes can be used to check for bottlenecks of the hardware.
+## Understanding the options and commands
 
-The duration of the benchmark can be set with `--duration` and defaults to 10 seconds. A longer duration will yield more accurate results.
+When running the profiler tool, you can provide several inputs to it. This section goes over the various commands and options for the profiler tool and what they mean.
 
-There are two parameters that can influence the proofing speed and can be optimized for go-spacemesh: `--threads` and `--nonces`.
-The general rule of thumb is the more threads the faster, but because different processors may have different behaviours when all threads are under load. Please check different values for `--threads` to find the optimal one.
-The more nonces are used with `--nonces` in ONE data pass, the bigger the chance to find a valid proof. The downside of using more `--nonces` is a heavier load on the CPU. It's important to note that nonces must be a multiple of 16.
+### Commands
 
-To simplify the estimation of nonces please use the following formula:
-in google sheets format
+* `proving`: We are telling the profiler to benchmark the entire PoST generation process.
+* `pow:`: We are telling the profiler to only benchmark the Proof of Work (PoW) part of the PoST generation. More on this below.
+
+If no command is given, the profiler runs the `proving` command by default.
+
+### Options
+
+* `--help`: Displays all the available commands and options along with their short descriptions. Use the `-h` flag for summarized help information.
+* `--data-file` (path): This is the path to an existing PoS file that we have already generated as a smesher during the [initialization or PoS generation](https://docs.spacemesh.io/docs/learn/post#generating-the-data) phase. By providing the path to an existing PoS data file, we are asking the profiler how long will it take to generate the PoST with the given PoS data. For accurate results, it is recommended to use the exact PoS data file on the same disk as used by the smeshing node. However, do note that the contents of the data file might get overwritten by the profiler tool.
+  * **Default**: The profiler tool will generate a 1 GiB PoS data file in a temporary directory.
+* `--data-size` (size in GiB): This is the size (in GiB) of the PoS data that we want the profiler to estimate the PoST generation time for. For example, we can provide a value of 5 GiB to have the profiler estimate the time it takes to generate the PoST given 5 GiB of PoS data.
+  * **Default**: A PoS data size of 1 GiB will be used for running the benchmark.
```

Comment:
This might give the impression that the users should provide a value matching their PoS size. It's not about it. The value should be big enough so that the benchmark is stable. With too small data size, the results might have a high variance.

> the profiler estimate the time it takes to generate the PoST given 5 GiB of PoS data.

The profiler doesn't estimate that time. It estimates the speed of proving in GiB/s. The user must divide the PoS size by this value to obtain the expected time manually.

Reply:
Reworded to the following:

`--data-size` (size in GiB): This is the size (in GiB) of the hypothetical PoS data that we want the profiler to estimate the PoST generation speed (in GiB/s) for. For example, we can provide a value of 5 GiB to have the profiler estimate the PoST generation speed given 5 GiB of PoS data. Once we have this speed value, we can divide our actual PoS data size by it to obtain the time (in seconds) of how long it would take our node to generate the PoST.

Also, I think we should enforce the data size to be in multiples of 64 (and have the default to be 64 GiB) as PoS is also in SUs. Wdyt?

Reply:
TBH, it sounds overcomplicated, I have trouble following what it is for. How about not going into too much detail about it? The user only needs to understand that increasing it gives more accurate results.

>Also, I think we should enforce the data size to be in multiples of 64 (and have the default to be 64 GiB) as PoS is also in SUs. Wdyt?

Then, running the profiler would require 64GiB of free disk space but the results wouldn't be much more accurate. I think it should stay 1GiB for easy math.

Reply:
This is not the size of the PoS data - the speed doesn't depend on the size. I'd stick with the latter.